spec/support/read_only.rb

```diff
@@ -0,0 +1,6 @@
+  # This allows us to unlock read only tables for testing
+  def set_read_only(klasses, readonly=false)
+    Array(klasses).each do |klass|
+      klass.define_method(:readonly?) { readonly }
+    end
```

Comment:
Might be better using mocks here as they'll get automatically torn down, removing the need for an after block, and reducing the risk of accidentally mutating the global state.

```ruby
klasses.each do |klass|
  allow_any_instance_of(Widget).to receive(:readonly?).and_return(readonly)
end
```
I think rubocop-rspec will complain, but I think its still a safer option.